### PR TITLE
contrib: add a simple planner that use lib/ai

### PR DESCRIPTION
**Pull request by privat**
This is a basic implementation for a TP given by a professor who wants to remain anonymous.
I'm quite sure the algorithm and the implementation can be improved, but it does the job in a quite acceptable time performance.

### CODE REVIEW

File: contrib/simplan/simplan.nit [hunk 1]
```diff
@@ -0,0 +1,629 @@
...
+
+	# Cost of loading a parcel in a robot
+	# Used for the action `Unload`
+	var duration_unloading = 30.0
+
+	# How many parcels can a robot transport at the same time?
+	var capacity = 2
+
+	# The name of the robot
+	var robot_name = "r0"
```
Comment (Morriar) [inline contrib/simplan/simplan.nit]: `r2d2` would have been a better name.

File: contrib/simplan/simplan.nit [hunk 2]
```diff
@@ -0,0 +1,629 @@
...
+	# Locations on the map, by their names
+	var locations = new HashMap[String, Location]
+
+	# All the parcels, by their `id`
+	var parcels = new Array[Parcel]
+
+	# All the parcels, by their names
+	var parcel_by_name = new HashMap[String, Parcel]
+
+	# A big number, usually +inf
```
Comment (Morriar) [inline contrib/simplan/simplan.nit]: Is this used when computing the shortest path? Documentation can be improved to explain why this is usefuull.

File: contrib/simplan/simplan.nit (hunk 1 above)
Comment (privat) [inline contrib/simplan/simplan.nit]: I will suggest it to the anonymous professor.

**Comment by privat** [comment]
reroll: improved documentation

**Comment by xymus** [comment]
+1

File: contrib/simplan/simplan.nit [hunk 3]
```diff
@@ -0,0 +1,639 @@
...
+
+	# The speed on the road.
+	# Used for the action `Drive`
+	var speed = 1.0
+
+	# Cost of loading a parcel in a robot
+	# Used for the action `Load`
+	var duration_loading = 30.0
+
+	# Cost of loading a parcel in a robot
```
Comment (R4PaSs) [inline contrib/simplan/simplan.nit]: :s/loading/unloading ?

File: contrib/simplan/simplan.nit [hunk 4]
```diff
@@ -0,0 +1,639 @@
...
+	var locations = new HashMap[String, Location]
+
+	# All the parcels, by their `id`
+	var parcels = new Array[Parcel]
+
+	# All the parcels, by their names
+	var parcel_by_name = new HashMap[String, Parcel]
+
+	# +infinite, used to initialize some floats with the maximum value.
+	private var infinite: Float = 1.0 / 0.0
```
Comment (R4PaSs) [inline contrib/simplan/simplan.nit]: Once #1612 is merged, this could disappear I guess

**Comment by R4PaSs** [comment]
:shipit:

**Pull request merged**